core/src/main/java/org/apache/iceberg/BaseMetastoreTableOperations.java

```diff
@@ -200,8 +200,13 @@ protected void refreshFromMetadataLocation(String newLocation, Predicate<Excepti
   private String metadataFileLocation(TableMetadata metadata, String filename) {
     String metadataLocation = metadata.properties()
         .get(TableProperties.WRITE_METADATA_LOCATION);
+    String metadataLocationSelectiveSuffix = metadata.properties()
+            .get(TableProperties.WRITE_METADATA_LOCATION_SELECTIVE_SUFFIX);
```

Comment:
May be I didn't explain my suggestion clearly. I was suggesting to reuse `TableProperties.WRITE_METADATA_LOCATION` and additionally add db-name and table-name to that location if a config is set. We can set `TableProperties.WRITE_METADATA_LOCATION` at catalog level.